pkg/util/deployment/deployment.go

```diff
@@ -367,6 +374,19 @@ func NewRSNewReplicas(deployment *extensions.Deployment, allRSs []*extensions.Re
 	}
 }
 
+// IsSaturated checks if the new replica set is saturated by comparing its size with its deployment size.
+func IsSaturated(deployment *extensions.Deployment, rs *extensions.ReplicaSet) bool {
```

Comment:
What saturated means in this wasn't immediately obvious to me when reading from the callers perspective, even though it does appear in a comment.  Usually I considered saturated as being full to the limit instead of having an exact count matching.  (e.g. a rate-limiter is saturated even if it is over the max amount)  As I understand it, this function returns true IFF both the deployment and the rs believe this rs can own all of the desired replicas in the deployment.  Is that correct?